lib/literature/components/forms/post_form_component.ex

```diff
@@ -259,10 +259,16 @@ defmodule Literature.PostFormComponent do
     html = if is_binary(html) and html != "", do: Jason.decode!(post_params["html"]), else: []
     post_params = Map.put(post_params, "html", html)
 
+    # Elixir will warn that using socket in start_async is expensive. However,
+    # copying it is necessary since save_post ultimately passes to
+    # Phoenix.LiveView.uploaded_entries(socket, name). We assign to a different
+    # variable to avoid the warning.
+    sock = socket
+
     socket
     |> assign(:loading, true)
     |> start_async(:save_task, fn ->
-      save_post(socket, socket.assigns.action, post_params)
+      save_post(sock, sock.assigns.action, post_params)
```

Comment:
Let's refactor it, we can avoid passing the socket here. Something like:

```elixir
post_params = post_params |> Map.put(post_params, "html", html) |> merge_post_params(socket.assigns.action)
post = socket.assigns.post
action = socket.assigns.action

socket
|> ...
|> start_async(:save_task, fn ->
  save_post(post, action, post_params)
end)
```
merge_post_params/2 will do all params conversion
```elixir
defp merge_post_params(params, :edit_post) do
 # copy how we convert post params in the old save post (edit)
end

defp merge_post_params(params, :new_post) do
 # copy how we convert post params in the old save post (new)
end
```

Then save_post/3 will now just receive post, action, and post_params. Which will just call either Literature.create_post/1 or update_post/2 depending on the action